website/src/common/Macrobench.tsx

```diff
@@ -42,197 +43,217 @@ export default function Macrobench({ data, gitRef, commits }) {
-              data.diff.Left.Metrics.TotalComponentsMemStatsAllocBytes
-            )}
-            right={formatByte(
-              data.diff.Right.Metrics.TotalComponentsMemStatsAllocBytes
-            )}
-            diffMetric={fixed(
-              data.diff.DiffMetrics.TotalComponentsMemStatsAllocBytes,
-              2
-            )}
+            oldVal={data.result.total_components_mem_stats_alloc_bytes.old}
+            newVal={data.result.total_components_mem_stats_alloc_bytes.new}
+            delta={data.result.total_components_mem_stats_alloc_bytes.delta}
+            insignificant={data.result.total_components_mem_stats_alloc_bytes.insignificant}
+            p={fixed(data.result.total_components_mem_stats_alloc_bytes.p, 3)}
+            fmt={"memory"}
           />
 
           <Row
             title="Allocated / query (vtgate)"
-            left={formatByte(
-              data.diff.Left.Metrics.ComponentsMemStatsAllocBytes.vtgate
-            )}
-            right={formatByte(
-              data.diff.Right.Metrics.ComponentsMemStatsAllocBytes.vtgate
-            )}
-            diffMetric={fixed(
-              data.diff.DiffMetrics.ComponentsMemStatsAllocBytes.vtgate,
-              2
-            )}
+            oldVal={data.result.components_mem_stats_alloc_bytes.vtgate.old}
+            newVal={data.result.components_mem_stats_alloc_bytes.vtgate.new}
+            delta={data.result.components_mem_stats_alloc_bytes.vtgate.delta}
+            insignificant={data.result.components_mem_stats_alloc_bytes.vtgate.insignificant}
+            p={fixed(data.result.components_mem_stats_alloc_bytes.vtgate.p, 3)}
+            fmt={"memory"}
           />
 
           <Row
             title="Allocated / query (vttablet)"
-            left={formatByte(
-              data.diff.Left.Metrics.ComponentsMemStatsAllocBytes.vttablet
-            )}
-            right={formatByte(
-              data.diff.Right.Metrics.ComponentsMemStatsAllocBytes.vttablet
-            )}
-            diffMetric={fixed(
-              data.diff.DiffMetrics.ComponentsMemStatsAllocBytes.vttablet,
-              2
-            )}
+            oldVal={data.result.components_mem_stats_alloc_bytes.vttablet.old}
+            newVal={data.result.components_mem_stats_alloc_bytes.vttablet.new}
+            delta={data.result.components_mem_stats_alloc_bytes.vttablet.delta}
+            insignificant={data.result.components_mem_stats_alloc_bytes.vttablet.insignificant}
+            p={fixed(data.result.components_mem_stats_alloc_bytes.vttablet.p, 3)}
+            fmt={"memory"}
           />
         </tbody>
       </table>
     </div>
   );
 }
 
-function Row({ title, left, right, diffMetric }) {
+export function getRange(range) {
+  if (range.infinite == true) {
+    return "∞"
+  }
+  if (range.unknown == true) {
+    return "?"
+  }
+  return "±"+fixed(range.value, 1)+"%"
+}
+
+function Row({ title, oldVal, newVal, delta, insignificant, p, fmt }) {
```

Comment:
This is nit, but since this is a tsx file anyway, could add types to this 😄

Reply:
This is a good point, I am not an expert of TypeScript but I gave it a try on the three `Row` functions we have: [10a00d9](https://github.com/vitessio/arewefastyet/pull/517/commits/10a00d970c2a6eaec3bf158a869a84876c9e664c), [dc4199f](https://github.com/vitessio/arewefastyet/pull/517/commits/dc4199f1b09e58dfa2551a5083778398e5e69ef0) and [058af50](https://github.com/vitessio/arewefastyet/pull/517/commits/058af50dce9554f8b8cb4f7dea40426c41309893)